File: scratch/test_nav.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
    const browser = await chromium.launch();
    const page = await browser.newPage();
    const indexPath = 'file://' + path.resolve(__dirname, '../index.html#courses?subject=cnxh&tab=quizzes');
    
    console.log('Navigating to:', indexPath);
    await page.goto(indexPath);
    await page.waitForTimeout(1000);

    // Take screenshot of course page
    await page.screenshot({ path: path.resolve(__dirname, '../scratch/test_courses.png') });
    console.log('Courses page screenshot saved.');

    // Click "Làm bài" on Chapter 1
    const lamBaiBtn = page.locator('text=Làm bài').first();
    await lamBaiBtn.click();
    await page.waitForTimeout(500);

    // Click "Bắt đầu làm bài" modal button
    const startBtn = page.locator('text=🚀 Bắt đầu làm bài');
    await startBtn.click();
    await page.waitForTimeout(1000);

    // Take screenshot of quiz room
    await page.screenshot({ path: path.resolve(__dirname, '../scratch/test_quiz_room.png') });
    console.log('Quiz room screenshot saved.');

    await browser.close();
    console.log('Test completed successfully!');
})();

File: scratch/test_nav.py
from playwright.sync_api import sync_playwright
import os

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    
    # 1. Test index.html
    index_path = 'file://' + os.path.abspath('index.html')
    print('Testing index.html:', index_path)
    page.goto(index_path)
    page.wait_for_timeout(1000)
    print('Title:', page.title())

    # Click "Khám phá Khóa học"
    btn = page.locator('text=Khám phá Khóa học').first
    btn.click()
    page.wait_for_timeout(1000)
    print('Navigated to:', page.url)

    # 2. Test courses.html
    print('Expanded subfolder...')
    subfolder = page.locator('.subfolder-header').first
    if subfolder.is_visible():
        subfolder.click()
        page.wait_for_timeout(500)

    # Click "Làm bài"
    lam_bai = page.locator('text=Làm bài').first
    lam_bai.click()
    page.wait_for_timeout(500)

    # Click "Bắt đầu làm bài"
    start_btn = page.locator('text=🚀 Bắt đầu làm bài').first
    start_btn.click()
    page.wait_for_timeout(1000)

    # 3. Test quiz-room.html
    print('Quiz room loaded at:', page.url)
    title_el = page.locator('#active-quiz-title')
    print('Active Quiz Title:', title_el.inner_text())

    # Take screenshot of quiz room
    page.screenshot(path='scratch/test_modular_quiz_room.png')
    print('Quiz room screenshot saved successfully!')

    browser.close()
